meterfs/meterfs.c

```diff
@@ -565,7 +567,8 @@ int meterfs_allocateFile(const char *name, size_t sectorcnt, size_t filesz, size
 	if (recordsz > filesz || !recordsz)
 		return -EINVAL;
 
-	strncpy(hdr.name, name, sizeof(hdr.name));
+	memset(hdr.name, 0, sizeof(hdr.name));
+	memcpy(hdr.name, name, err);
```

Comment:
I changed it to use lenght value returned from rewritten `meterfs_fileNameCheck()` function. I added `memset()` because originally `strncpy()` fulfill rest of buffer with zeros.

Reply:
Yeah, it would require additional math, not really worth on such short array.